examples/universal-cnf/vppagent/pkg/config/config_endpoints.go

```diff
@@ -44,8 +44,14 @@ type CompositeEndpointAddons interface {
 	AddCompositeEndpoints(*common.NSConfiguration, *nseconfig.Endpoint) *[]networkservice.NetworkServiceServer
 }
 
+type IpamService interface {
+	AllocateSubnet(ucnfEndpoint *nseconfig.Endpoint) (string, error)
+}
+
+type IpamServiceFactory func(addr string) IpamService
+
 // NewProcessEndpoints returns a new ProcessInitCommands struct
-func NewProcessEndpoints(backend UniversalCNFBackend, endpoints []*nseconfig.Endpoint, nsconfig *common.NSConfiguration, ceAddons CompositeEndpointAddons) *ProcessEndpoints {
+func NewProcessEndpoints(backend UniversalCNFBackend, endpoints []*nseconfig.Endpoint, nsconfig *common.NSConfiguration, ceAddons CompositeEndpointAddons, newService IpamServiceFactory) *ProcessEndpoints {
```

Comment:
`newService` seems like a bad name here, it's not really telling about what kind of service it is exposing.

Reply:
done